torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -991,12 +982,33 @@ def _get_params_per_wrapped_module(self, root_module) -> List[List[nn.Parameter]
             ignored_params=[],
             only_wrap_children=False,
             )
-        # parameters not wrapped in ``policy`` are grouped together.
-        remaining_params = [
-            p for p in root_module.parameters() if p not in traversed_param_set
-        ]
-        params_per_wrapped_module.append(remaining_params)
-        return params_per_wrapped_module
+        if root_module not in modules_to_wrap:
+            modules_to_wrap.append(root_module)
+
+        params_per_wrap: List[List[nn.Parameter]] = []
+        traversed_params: Set[nn.Parameter] = set()
+        # Modules in `modules_to_wrap` are ordered based on post-order traversal.
```

Comment:
nit: 
```suggestion
        # Constructing `modules_to_wrap` with `_recursive_wrap` orders the modules based on a post-order traversal.
```
I just wanted to clarify that this post-order traversal comes from `_recursive_wrap`.

That being said, maybe we should mention in the `_recursive_wrap` docstring that it is post-order if not already since now we are relying on that.

Reply:
not relevant to this PR, but I think it's better to change the name of `_recursive_wrap` to something like `_wrap_post_order_traversal`. It's not clear what `_recursive_wrap` means